doc/tutorials/using_cat_on_up2.rst

```diff
@@ -0,0 +1,157 @@
+.. _using_cat_up2:
+
+Using CAT on UP2
+################
+
+The UP Squared board `(UP2) <https://up-board.org/upsquared/specifications/>`_
+is an x86 maker board based on the Intel Apollo Lake platform, which support
+Cache Allocation Technology(CAT). With this feature, the usage of cache can be
+restricted to each VM. ACRN hypervisor can do that by giving VM a cache mask,
+so that the VM will not evict the masked cache lines, as shown in :numref:
+`cache-mask-vm`:
+
+.. figure:: images/using_cat_up2.png
+   :align: center
+   :name: cache-mask-vm
+
+   Cache usage with Cache Mask
+
+
+CAT Support on ACRN
+*******************
+
+As described at ``Intel® 64 and IA-32 Architectures Software Developer’s Manual``,
+chapter 17.17, volume 3. There are 3 steps to use CAT: detecting the CAT capability;
+setup cache mask array MSRs, which is refered as Class of Service(CLOS) array; select
+one of the CLOS array for the CPU, that will be the cache mask of the CPU.
+ACRN integrated the usage of CAT into vm configuration. The CLOS array must be defined
+at the board config source code, and the VM need to specify which CLOS to use in its vm
+config data. If the platform has CAT, the CLOS array will be write to CLOS MSRs at cpu
+init time, and the VM will set CLOS for VCPU at VM launch time.
+The details is shown in :numref:`acrn-cat-hld`:
+
+.. figure:: images/acrn_cat_hld.png
+   :align: center
+   :name: acrn-cat-hld
+
+   CAT Support on ACRN
+
+Tuning CAT at HV debug shell
+############################
+
+When at ACRN HV debug shell, user can use 'cpuid', 'wrmsr/rdmsr' debug commands to
+enumarate CAT capability and tune CAT parameters. We will use UP2 board as an example,
+the UP2 can use its serial port for HV shell, how to setup UP2 serial port please refer
+to `Getting started guide for UP2 board <https://projectacrn.github.io/latest/getting-started/up2.html>`
+
+#. Check CAT ability with cupid, first run 'cpuid 0x10 0x0', the return value ebx[bit 2]
+   report the L2 CAT is supported. Then run 'cpuid 0x10 0x2' to query L2 CAT capability,
+   the return eax[bit 4:0] report cache mask has 8 bit, edx[bit 15:0] report 4 CLOS are
+   supported
+
+   .. code-block:: none
+
+     ACRN:\>cpuid 0x10 0x0
```

Comment:
I believe this block needs to be aligned with the 'c' from `code-block`

Reply:
By convention, the code block contents should be indented as Geoffroy mentions, but it will work as written.